instrumentation/apache-httpclient/apache-httpclient-4.0/src/main/java/io/opentelemetry/auto/instrumentation/apachehttpclient/v4_0/ApacheHttpClientInstrumentation.java

```diff
@@ -172,16 +171,14 @@ public ApacheHttpClientInstrumentation() {
 
   public static class HelperMethods {
     public static SpanWithScope doMethodEnter(final HttpUriRequest request) {
-      final Span span =
-          TRACER.spanBuilder(DECORATE.spanNameForRequest(request)).setSpanKind(CLIENT).startSpan();
+      final Span span = DECORATE.getOrCreateSpan(request, TRACER);
 
       DECORATE.afterStart(span);
       DECORATE.onRequest(span, request);
 
-      final Context context = withSpan(span, Context.current());
-      final boolean awsClientCall = request.getHeaders("amz-sdk-invocation-id").length > 0;
-      // AWS calls are often signed, so we can't add headers without breaking the signature.
-      if (!awsClientCall) {
```

Comment:
Now that there's only the SDK span and no Apache span, there's nothing to propagate here. That being said, it does seem safer to keep the check just in case something changes, at the same time, it seems pretty hacky to have the AWS-specific code here so it feels nice to remove it.

Reply:
Oh, I understand now, right.

Do the `aws-sdk` tests protect us here? If so I'm good removing it.

Reply:
Yup the tests cover it. It's how I found this line of code :)